susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+                    for line in f.read().splitlines():
+                        print(line)
+        raise RuntimeError(
+            "Upgrade failed! Most probably the new SQL scripts are not idempotent!")
+
+
+def diff_dumps(initial_dump, migrated_dump):
+    fileInitial = open(initial_dump)
+    fileMigrated = open(migrated_dump)
+    return(difflib.context_diff(fileInitial.readlines(), fileMigrated.readlines()))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Test idempotency of SQL scripts")
+    parser.add_argument("-s", "--schema-path", action="store", dest="schemaPath",
+                        help="Path where the directories with the schema upgrades are (default: /etc/sysconfig/rhn/schema-upgrade",
+                        default="/etc/sysconfig/rhn/schema-upgrade")
+    parser.add_argument("-p", "--from-pr", action="store", dest="prFile",
+                        help="Check files with changes from a PR by rerunning added or changed files. The value is the path to a JSON file with PR information generated by gitarro")
+    parser.add_argument("-v", "--from-version", action="store", dest="version",
+                        help="Check all files since a base version (for example 4.0.0)")
+    parser.add_argument("-u", "--upgrade-script", action="store", dest="upgradeScript",
+                        help="Path where the spacewalk-schema-upgrade script is (default: /manager/schema/spacewalk/spacewalk-schema-upgrade)",
+                        default="/manager/schema/spacewalk/spacewalk-schema-upgrade")
+    args = parser.parse_args()
+    if args.schemaPath is None:
+        raise RuntimeError(
+            "You need to specify the path where the schema upgrades are")
+    if not directory_exists(args.schemaPath):
+        raise RuntimeError("Directory %s does not exist" % args.schemaPath)
+    if args.prFile is None and args.version is None:
+        raise RuntimeError(
+            "One of --from-pr or --from-version must be specified")
+    if args.prFile and args.version:
+        raise RuntimeError(
+            "Only one of --from-pr or --from-version must be specified")
+    if args.prFile and not file_exists(args.prFile):
+        raise RuntimeError("File %s does not exist" % prFile)
+    if args.upgradeScript is None:
+        raise RuntimeError(
+            "You need to specify the path where the spacewalk-schema-upgrade script is")
+    if not file_exists(args.upgradeScript):
+        raise RuntimeError(
+            "The file %s for the upgrade script does not exist" % args.upgradeScript)
+    print("============================================================")
+    print("                     Idempotency test                       ")
+    print("============================================================")
+    # Exclude tables and sequences that are not updated by the SQL scripts (version control)
+    excludedTables = ["pg_catalog.setval", "public.rhnpackageevr",
```

Comment:
i would personally put them outside of main and pass them as ARG to main. 

If python has constant like golang/ruby i would freeze them